### PR TITLE
feat/reset password ui

### PR DESCRIPTION
**Pull request by alexworker23**
Screenshot 2021-11-22 at 11 17 26
Screenshot 2021-11-22 at 11 17 55

### CODE REVIEW

File: packages/ui/libs/device/data/src/fetching/deviceFirstImage.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ export const useDeviceFirstImageUrl = (
   const [imageUrl, setImageUrl] = useState('');
 
   const getAndSetImage = async (id: string) => {
+    if (id === 'null') return;
+
     const response = await publicFileDownloadHandler(id);
```
Comment (ioncreature) [inline packages/ui/libs/device/data/src/fetching/deviceFirstImage.ts]: Looks a bit strange. Is it really `"null"` string? If yes, maybe it is better to fix somewhere before this happens
The same for `packages/ui/libs/device/data/src/fetching/deviceImages.ts:11`

Comment (alexworker23) [reply packages/ui/libs/device/data/src/fetching/deviceFirstImage.ts]: I agree, this is how i received it from `/api/device-registry` GET:
Screenshot 2021-11-25 at 09 59 13

Comment (ioncreature) [reply packages/ui/libs/device/data/src/fetching/deviceFirstImage.ts]: So this is definitely a bug on backend, I'll check what is going on there

**Comment by Malak67** [comment]
Is there a key/token which will be added in the URL received in the email which then redirects to Change password?
I'm just wondering because I did not see a way to identify the account which is changing the password. But maybe you're waiting for the backend to be done.

**Comment by alexworker23** [comment]
> Is there a key/token which will be added in the URL received in the email which then redirects to Change password? I'm just wondering because I did not see a way to identify the account which is changing the password. But maybe you're waiting for the backend to be done.

Yes, there will be a token, but the backend implementation for this is not done yet. That's why I left submitHandlers as empty functions with comments above them.

File: packages/ui/libs/device/data/src/fetching/deviceFirstImage.ts (hunk 1 above)
Comment (ioncreature) [reply packages/ui/libs/device/data/src/fetching/deviceFirstImage.ts]: it was a bug on the backend, added a task to validate imageIds.

**Review by ioncreature — APPROVED** [review]

**Review by Malak67 — APPROVED** [review]

**Review by driamend — APPROVED** [review]

**Pull request merged**